acmeclient.go

```diff
@@ -144,15 +144,15 @@ func (am *ACMEManager) newACMEClient(ctx context.Context, useTestCA, interactive
 	// in situations where the default challenges would fail)
 	if am.DNS01Solver == nil {
 		// enable HTTP-01 challenge
-		if !am.DisableHTTPChallenge {
+		if !am.DisableHTTPChallenge && am.HTTP01Solver == nil {
 			useHTTPPort := HTTPChallengePort
 			if HTTPPort > 0 && HTTPPort != HTTPChallengePort {
 				useHTTPPort = HTTPPort
 			}
 			if am.AltHTTPPort > 0 {
 				useHTTPPort = am.AltHTTPPort
 			}
-			client.ChallengeSolvers[acme.ChallengeTypeHTTP01] = distributedSolver{
+			am.HTTP01Solver = distributedSolver{
```

Comment:
On second thought; instead of setting the exported field to a non-nil value here, can we set a local variable instead? I just want to avoid drastic changes to the `am` struct, it might surprise users after a method call that suddenly this field is non-nil.

(Same below)